python/pyarrow/_dataset.pyx

```diff
@@ -269,20 +454,21 @@ cdef class FileSystemDataset(Dataset):
     cdef:
         CFileSystemDataset* filesystem_dataset
 
-    def __init__(self, paths_or_selector, schema=None, format=None,
-                 filesystem=None, partitions=None, root_partition=None):
+    def __init__(self, paths_or_selector, schema, format, filesystem,
+                 partitions=None, Expression root_partition=_true):
```

Comment:
Can you leave this as it was? This was done on purpose to provide decent error messages (see discussion in https://github.com/apache/arrow/pull/6913, I am happy to find a better way though)

Reply:
reverted